contracts/deploy/mainnet/127_replace_dai_with_usds.js

```diff
@@ -0,0 +1,208 @@
+      "initialize()",
+      []
+    );
+    const initFunction = "initialize(address,address,bytes)";
+    await withConfirmation(
+      cMakerSSRStrategyProxy.connect(sDeployer)[initFunction](
+        dMakerSSRStrategy.address,
+        timelockAddr, // governor
+        initData, // data for delegate call to the initialize function on the strategy
+        await getTxOpts()
+      )
+    );
+
+    return {
+      name: "Replace DAI with USDS",
+      actions: [
+        {
+          // Upgrade VaultAdmin implementation
+          contract: cVault,
+          signature: "setAdminImpl(address)",
+          args: [dVaultAdmin.address],
+        },
+        {
+          // Set OracleRouter on Vault
+          contract: cVault,
+          signature: "setPriceProvider(address)",
+          args: [cOracleRouter.address],
+        },
+        {
+          // Support USDS on Vault
+          contract: cVault,
+          signature: "supportAsset(address,uint8)",
+          args: [USDS, 0],
+        },
+
+        {
+          // Approve Migration Strategy on the Vault
+          contract: cVault,
+          signature: "approveStrategy(address)",
+          args: [migrationStrategy.address],
+        },
+        {
+          // Approve SSR Strategy on the Vault
+          contract: cVault,
+          signature: "approveStrategy(address)",
+          args: [cMakerSSRStrategy.address],
+        },
+        {
+          // Set Migration Strategy as default for DAI
+          contract: cVault,
+          signature: "setAssetDefaultStrategy(address,address)",
+          args: [DAI, migrationStrategy.address],
+        },
+        {
+          // Set Maker SSR Strategy as default for USDS
+          contract: cVault,
+          signature: "setAssetDefaultStrategy(address,address)",
+          args: [USDS, cMakerSSRStrategy.address],
+        },
+        {
+          // Remove DSR strategy
+          // This would move all DAI to vault
+          contract: cVault,
+          signature: "removeStrategy(address)",
+          args: [cDSRStrategyProxy.address],
+        },
+        {
+          // Allocate DAI from the Vault to the Migration Strategy
+          contract: cVault,
+          signature: "allocate()",
+          args: [],
+        },
+        {
+          // Reset default strategy for DAI so that we can remove it
+          contract: cVault,
+          signature: "setAssetDefaultStrategy(address,address)",
+          args: [DAI, addresses.zero],
+        },
+        {
+          // Remove Migration Strategy
+          // This will remove all USDS and move it to the Vault
+          contract: cVault,
+          signature: "removeStrategy(address)",
+          args: [migrationStrategy.address],
+        },
+
+        // {
+        //   // Optional: Allocate USDS from the Vault to the SSR Strategy
+        //   contract: cVault,
+        //   signature: "allocate()",
+        //   args: [],
+        // },
```

Comment:
I would say that I'm favor of allocating all the USDS from the vault to the SSR strategy. Atomic change are easier to understand.

Reply:
The thing is it doesn't have to be part of the governance proposal, we can manually trigger the allocate after the proposal is executed. I commented it out since the proposal is getting huge 😅